### ISSUE TITLE
SC1087: false positive, array detected instead a regex

### ISSUE THREAD

**Issue by kyodev** [koalaman/shellcheck#1259]
#### For bugs

- SC1087
- My shellcheck version 0.5.0
- [x ] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

ireq="test"
file="test         install          bla"

grep -E "^$ireq[[:space:]]+install" <<< "$file"
```

#### Here's what shellcheck currently says:

```txt
grep -E "^$ireq[[:space:]]+install" <<< "$file"
          ^-- SC1087: Use braces when expanding arrays, e.g. ${array[idx]} (or ${var}[.. to quiet).
```

#### Here's what I wanted or expected to see:

```txt
test         install          bla
```
no error, it is not an array, just a regex in grep with a variable

**Comment by kyodev** [comment]
sorry, I readed correctly
